Tests/MaterialLib/TestFluidDensity.cpp

```diff
@@ -0,0 +1,130 @@
+using namespace MaterialLib::Fluid;
+
+//----------------------------------------------------------------------------
+// Test density models.
+std::unique_ptr<FluidProperty> createTestFluidDensityModel(const char xml[])
+{
+    auto const ptree = readXml(xml);
+    BaseLib::ConfigTree conf(ptree, "", BaseLib::ConfigTree::onerror,
+                             BaseLib::ConfigTree::onwarning);
+    auto const& sub_config = conf.getConfigSubtree("density");
+    return MaterialLib::Fluid::createFluidDensityModel(sub_config);
+}
+
+TEST(Material, checkConstantDensity)
+{
+    const char xml[] =
+        "<density>"
+        "   <type>Constant</type>"
+        "   <value> 998.0 </value> "
+        "</density>";
+    const auto rho = createTestFluidDensityModel(xml);
+
+    ASSERT_EQ(998.0, rho->getValue(nullptr));
+    ASSERT_EQ(0.0,
+              rho->getdValue(nullptr, MaterialLib::Fluid::PropertyVariable::T));
+}
+
+TEST(Material, checkIdealGasLaw)
+{
+    const char xml[] =
+        "<density>"
+        "   <type>IdealGasLaw</type>"
+        "   <molar_mass> 28.96 </molar_mass> "
+        "</density>";
+    const auto rho = createTestFluidDensityModel(xml);
+
+    const double molar_air = 28.96;
+    const double T = 290.;
+    const double p = 1.e+5;
+    const double R = PhysicalConstant::IdealGasConstant;
+    const double expected_air_dens = molar_air * p / (R * T);
+    double vars[] = {290, 0, 1.e+5};
+    ASSERT_NEAR(expected_air_dens, rho->getValue(vars), 1.e-10);
+
+    const double expected_d_air_dens_dT = -molar_air * p / (R * T * T);
+    ASSERT_NEAR(expected_d_air_dens_dT,
+                rho->getdValue(vars, Fluid::PropertyVariable::T), 1.e-10);
+
+    const double expected_d_air_dens_dp = molar_air / (R * T);
+    ASSERT_NEAR(expected_d_air_dens_dp,
+                rho->getdValue(vars, Fluid::PropertyVariable::pg), 1.e-10);
+}
+
+TEST(Material, checkLinearTemperatureDependentDensity)
+{
+    const char xml[] =
+        "<density>"
+        "   <type>TemperatureDependent</type>"
+        "   <temperature0> 293.0 </temperature0> "
+        "   <beta> 4.3e-4 </beta> "
+        "   <rho0>1000.</rho0>"
+        "</density>";
+
+    const auto rho = createTestFluidDensityModel(xml);
+
+    double vars[] = {273.1 + 60.0};
+    ASSERT_NEAR(1000.0 * (1 + 4.3e-4 * (vars[0] - 293.0)), rho->getValue(vars),
+                1.e-10);
+    ASSERT_NEAR(1000.0 * 4.3e-4,
+                rho->getdValue(vars, Fluid::PropertyVariable::T), 1.e-10);
+}
+
+TEST(Material, checkLiquidDensity)
+{
+    const char xml[] =
+        "<density>"
+        "   <type>LiquidDensity</type>"
+        "   <temperature0> 273.15 </temperature0> "
+        "   <p0> 1.e+5 </p0> "
+        "   <bulk_modulus> 2.15e+9 </bulk_modulus> "
+        "   <beta> 2.0e-4 </beta> "
+        "   <rho0>999.8</rho0>"
+        "</density>";
+    const auto rho = createTestFluidDensityModel(xml);
+
+    const double vars[] = {273.15 + 60.0, 1.e+6};
+    const double T0 = 273.15;
```

Comment:
It is a reference temperature from the input data (in the xml array in this test). Therefore I did not use the constant by PhysicalConstant::CelsiusZeroInKelvin.